feat(github-actions): support GitHub actions runners (#23633)
renovatebot/renovate: lib/modules/datasource/github-runners/readme.md

This datasource returns a list of all _stable_ runners that are hosted by GitHub.
This datasource ignores beta releases.
The datasource is based on [GitHub's `runner-images` repository](https://github.com/actions/runner-images).

Examples: `windows-2019` / `ubuntu-22.04` / `macos-13`

## Maintenance

New _stable_ runner versions must be added to the datasource with a pull request.

> Unstable runners are tagged as `[beta]` in the readme of the [`runner-images` repository](https://github.com/actions/runner-images). Once a runner version becomes stable, the `[beta]` tag is removed and the suffix `latest` is added to its YAML label.

Reviewer: ```suggestion
Unstable runners are tagged as `[beta]` in the readme of the [`runner-images` repository](https://github.com/actions/runner-images).
Once a runner version becomes stable, the `[beta]` tag is removed and the suffix `latest` is added to its YAML label.
```

We follow the one sentence per line pattern in Markdown files.

Author: Is there a style guide for those rules?

Reviewer: Good question! You're a new contributor, so I guess some of my style changes feel arbitrary.

I created a style guide a while back, but forgot to add the "Use one sentence per line" for Markdown files rule to it. I'll make a PR to update the guide now. 😉 

Here's our [style guide](https://github.com/renovatebot/renovate/blob/main/docs/development/style-guide.md).

This datasource returns a list of all available runners hosted by GitHub based on `https://github.com/actions/runner-images`. Only stable runners are returned (no beta releases).

Examples: `windows-2019` / `ubuntu-22.04` / `macos-13`

**Maintenance**

> New runner versions have to be added manually to the data source via pull request.

Reviewer: ```suggestion
New _stable_ runner versions must be added to the datasource with a pull request.
```
